campaigns/workflows.md

```diff
@@ -0,0 +1,345 @@
+  user can modify an existing Notebook and return to that Notebook later...and
+  possibly share *it* via the same service.
+- A service that informs the user of all of their scattered persistent Binder
+  projects and can take them "back" to their persistent Binder with a single
+  click.
+- A caching service for selected objects in a shared Notebook, allowing users to
+  share *parts* of their Notebook in a way that can be directly used in other
+  Notebooks.  For example, a user could decide to share their Notebook via the
+  above service, at which point the user is prompted to fill out a form that
+  provides a basic description of the Notebook *and* asks the user to (optionally)
+  select objects in the Notebook they would like to share as part of the Notebook's
+  "public API".
+- ...Ideally, all of the above projects could be implemented and integrated with
+  each other.
+
+### Ingestion
+
+The *Ingestion* phase describes the step in a scientific workflow when the
+user loads the data they want into an analysis-ready format, such as an Xarray
+Dataset.  This phase should abstract away concepts that the user should not need
+to know about the data itself, such as the format of the data and where the data
+is located.  Ideally, the user should only need to understand the semantics
+describing the scientific aspects of the data that actually impact their analysis.
+Data format or location should not be completely *hidden* from the user, but they
+should not need to know anything about that to perform analysis.
+
+Ideally, a user would simply "select a dataset"
+(see [Search & Discovery](#search--discovery))
+and be provided with a queriable object, such as an Intake catalog object.  The
+user should be able to search and subselect this object using terminology that is
+understood by the *users of the data*, not the developers of the software.
+Ideally, the queriable catalog object should provide the user with the
+information they need to understand what the data *is*, rather than needing the
+user to look up information on an external website.  Ideally, once the queriable
+object has been queried (possibly multiple times) to subselet the data desired
+for the user's analysis, it should be trivial to return the subselected dataset
+in an analysis-ready data format, such as an Xarray Dataset.
+
+Much of this is already provided by `intake` and its plugins, but a critical
+aspect of this step is making it easy for data *providers* or *managers* (i.e.,
+the people who make the data available for others to analyze) to create catalogs
+with the proper semantic translations.  If this step (catalog generation) is
+difficult, then catalogs will not be created *at all*, making all the effort of
+[Phase 1](#search--discovery) pointless.  Ideally, catalogs should be generated
+automatically when data is generated, or they should be easily generated
+semi-automatically with a simple command-line utility.
```

Comment:
I thought ingestion should include a process that converts irregular grid data to regular grid data, which can be used to plot the data correctly. This step is very important and should be separated from visualization step.

Reply:
Regridding belongs in the *analysis* phase.